keps/sig-storage/2848-transfer-volumesnapshots-namespaces/README.md

```diff
@@ -0,0 +1,795 @@
+nitty-gritty.
+-->
+
+`VolumeSnapshotTransferRequest` resource:
+
+```yaml
+apiVersion: v1alpha1
+kind: VolumeSnapshotTransferRequest
+metadata:
+  name: transfer-foo
+  namespace: target-namespace
+spec:
+  source:
+    name: foo
+    namespace: source-namespace
+    kind: VolumeSnapshot
+  approvalName: approve-foo
+  targetName: bar
+```
+
+`VolumeSnapshotTransferApproval` resource:
+
+```yaml
+apiVersion: v1alpha1
+kind: VolumeSnapshotTransferApproval
+metadata:
+  name: approve-foo
+  namespace: source-namespace
+spec:
+  source:
+    name: foo
+    kind: VolumeSnapshot
+  targetNamespace: target-namespace
+  requestName: transfer-foo
+```
+
+When matching resources are detected, the controller initiate the transfer process.
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+Users should be able to transfer VolumeSnapshots that reference the same VolumeSnapshotContent
+across namespaces. This allows users to quickly share snapshots, utilizing data for testing
+or backups in separate namespaces.
+#### Story 1
+
+- The `prod` namespace contains a PVC that holds user information. VolumeSnapshots are taken of
+this PVC on a regular basis.
+
+- An existing VolumeSnapshot can be transferred to a test namespace. The user initiates the request
+by creating a `VolumeSnapshotTransferRequest` in the source namespace.
+
+- Once created, a `VolumeSnapshotTransferApproval` resource in the target namespace will be created.
+If matching resources are found, the VolumeSnapshot controller can init the transfer. 
+
+- When the transfer is successful, the original VolumeSnapshot should be deleted to assure there is one-to-one mapping between VolumeSnapshot and VolumeSnapshotContent.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+This KEP proposes the minimal to transfer VolumeSnapshot between namespaces, while avoiding breaking the current API.
+Many caveats have been discussed, including but not limited to :- 
+
+- [StorageClass Secrets](https://kubernetes-csi.github.io/docs/secrets-and-credentials-storage-class.html).
+- VolumeSnapshotContent.Spec.VolumeSnapshotRef is `immutable`. 
```

Comment:
This should not be a problem.  Just need to create a new VolumeSnapshotContent pointing to the same snapshot handle.

Reply:
@xing-yang removed 🙏🏽